src/app/shared/entity/opa.ts

```diff
@@ -0,0 +1,112 @@
+export class ConstraintTemplate {
```

Comment:
Not sure yet, if we need everything from the original spec, but therefor everything is in for now.